EX2/6.5_group_by.py

```diff
@@ -0,0 +1,24 @@
+def group_by(func, iterable):
```

Comment:
Use type hints and return types -
https://www.geeksforgeeks.org/duck-typing-in-python/